README.md

```diff
@@ -31,167 +31,119 @@ Where `<snapshot version>` refers to the version in the Sonatype Snapshot badge
 ```
 
-### Fluent API
+### Bulkable
 
-Both Elastic requests and queries offer a fluent API, so that you can provide optional parameters in chained method calls for each request or query. For example, if we wanted to add routing and refresh parameters to a `deleteById` request:
+Elastic Requests like `Create`, `CreateOrUpdate`, `CreateWithId`, `DeleteById` are bulkable requests. For bulkable request you can use `bulk` API that accepts request types that inherit `Bulkable` trait.
 
 ```scala
-deleteById(IndexName("index"), DocumentId("documentId")).routing(Routing("routing")).refreshTrue
+for {
+  _ <- Elasticsearch.execute(bulk(requests: _*)) 
+} yield()
 ```
 
-Just like `IndexName`, `Routing` is a new type that mustn't be an empty string.
+### Usage of ZIO Schema and its accessors for type safety
 
-And if we wanted to specify lower and upper bounds for a `range` query:
+To provide type safety in your requests zio-elasticsearch uses ZIO Schema. Here is an example of creating schema for custom type `User` and using implicit schema to create accessors which results in type safe request and response.
 
 ```scala
-range(EmployeeDocument.age).gte(18).lt(100)
-```
+final case class Address(street: String, number: Int)
 
-### Elastic Executor
+object Address {
 
-In order to get the functional effect of executing a specified Elasticsearch request, you must call the `execute` method defined on it, which returns a `ZIO` that requires an `ElasticExecutor`, fails with a `Throwable` and returns the relevant value `A` for that request.
+  implicit val schema: Schema.CaseClass2[String, Int, Address] =
+    DeriveSchema.gen[Address]
 
-Elastic requests for creating and deleting return `CreationOutcome` and `DeletionOutcome` respectively if no other meaningful value could be returned, notifying us on the success of the request. Any other kind of error is returned as a `Throwable` in the error channel of `ZIO` for that Elastic request.
+  val (street, number) = schema.makeAccessors(FieldAccessorBuilder)
+}
 
-If you want to execute multiple Elasticsearch requests in a single API call, you need to use the `bulk` method on those Elastic requests, and call `execute` on that bulk request instead.
+case class User(id: Int, address: Address)
 
-To provide the dependency on `ElasticExecutor`, you must pass one of the `ZLayer`s from the following import:
+object User {
+  implicit val schema: Schema.CaseClass2[String, Address, User] =
+    DeriveSchema.gen[User]
 
-```scala
-import zio.elasticsearch.ElasticExecutor
-```
+  val (id, address) = schema.makeAccessors(FieldAccessorBuilder)
+}
 
-For example, if you want to execute requests on a server running on `localhost` and port `9200`, you can provide the `live` ZLayer to your effect, along with a `SttpBackend` and an `ElasticConfig` layer:
+for {
+  _ <- Elasticsearch.execute(
+    search(
+      IndexName("index-name"),
+      must(range(User.id).gte(7).lt(10)).should(startsWith(User.address.name, "ZIO"))
```

Comment:
Let's use `User.address.streetName`. However, I'm not sure if is it a valid query in Elasticsearch.